plugin/wp-cli-login-server.php

```diff
@@ -135,6 +135,9 @@ public function run()
             $user  = $this->validate($magic);
             $this->loginUser($user);
             $this->loginRedirect($user, $magic->redirect_url);
+        } catch (AuthenticationFailure $e) {
+            $this->deleteMagic();
+            $this->abort($e);
         } catch (Exception $e) {
             $this->abort($e);
```

Comment:
Probably best to also delete on a generic exception and mention that in the user message.